### ISSUE TITLE
Object3d.getBoundingBox() returning null. Calling java.nio package methods which shouldn't be called

### ISSUE THREAD

**Issue by JehandadK** [Rajawali/Rajawali#1774]
#### Rajawali Version or Branch

stable release org.rajawali3d:rajawali:1.0.325@aar
#### Device and Android Version

Google Tango Device API 19 Release Vega
#### Summary

Cannot render a bounding box around a 3dObject loaded from obj file. While trying getBoundingBox a Nullpointer is thrown
#### Steps to Reproduce

```
        Log.d(TAG, mObject.getGeometry() + "");
        Log.d(TAG, mObject.getGeometry().getBoundingBox() + ""); // Error thrown here
        mObject.setShowBoundingVolume(true);
```
#### Trace or Log Output

```
java.lang.NullPointerException: Attempt to invoke virtual method 'java.nio.Buffer java.nio.FloatBuffer.rewind()' on a null object reference
                      at org.rajawali3d.bounds.BoundingBox.calculateBounds(BoundingBox.java:143)
                      at org.rajawali3d.bounds.BoundingBox.<init>(BoundingBox.java:73)
                      at org.rajawali3d.Geometry3D.getBoundingBox(Geometry3D.java:1011)
                      at com.royalcyber.ontango.SceneRenderer.onModelLoadComplete(SceneRenderer.java:110)
                      at org.rajawali3d.renderer.RajawaliRenderer$19.handleMessage(RajawaliRenderer.java:1046)
                      at android.os.Handler.dispatchMessage(Handler.java:102)
                      at android.os.Looper.loop(Looper.java:136)
                      at android.app.ActivityThread.main(ActivityThread.java:5030)
                      at java.lang.reflect.Method.invoke(Native Method)
                      at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:793)
                      at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:609)
```

Related to: http://stackoverflow.com/questions/39522747/rendering-a-frame-around-a-3d-object-using-opengl/39523907?noredirect=1#comment66363414_39523907

**Comment by jwoolston** [comment]
This crash happened because the geometry you are trying to calculate a bounding box for does not appear to have a vertex buffer.

**Comment by JehandadK** [comment]
How can I create it?

**Comment by jwoolston** [comment]
Is this a child object? If so, it is expected, you will need to get the bounding box from the parent.

**Comment by JehandadK** [comment]
Its the object loaded using LoaderObj. the return paremeter from LoaderObj.getParserdObject()

**Comment by jwoolston** [comment]
In that case it is not expected so it seems this is a legitimate bug, but I will be honest, I am not likely to deal with this due to the major structural changes I am working on with the 2.0 overhaul. I will however accept fixes and help anyone who wants to attempt to develop a fix.

**Comment by JehandadK** [comment]
I am interested but I am a beginner in opengl, not in java. I will need direction from you.

**Comment by jwoolston** [comment]
The issue is purely in the java, no open gl knowledge necessary. The first step would be to look at the parsed object graph and try to figure out why the vertex buffer is null. My initial thought was that it was a container object with no geometry but that can't be the case or the NPE would have happened earlier in the call stack.

**Comment by JehandadK** [comment]
I have found that the LoderOBJ examples for rajawali master branch is creating bounding boxes for its children correctly. Only parent does not have a bounding box. So using the code for creating bounding box for children I did this.

```java
public BoundingBox createBoundingBox(Object3D root) {
            Vector3 min = new Vector3();
            Vector3 max = new Vector3();

            min.setAll(Double.MAX_VALUE, Double.MAX_VALUE, Double.MAX_VALUE);
            max.setAll(-Double.MAX_VALUE, -Double.MAX_VALUE, -Double.MAX_VALUE);

            calFulLGeometry(root, min, max);

            return new BoundingBox(calculatePoints(min, max));
        }

private void calFulLGeometry(Object3D root, Vector3 min, Vector3 max) {
            for (int i = 0; i < root.getNumChildren(); i++) {
                if (root.getChildAt(i).getNumChildren() > 0)
                    calFulLGeometry(root.getChildAt(i), min, max);
                else
                    calGeometry(min, max, root.getChildAt(i));
            }
        }
```

Full gist [here](https://gist.github.com/JehandadK/6b74935cd2d24cb25d6f7889ecc34fe8).

Which correctly gives me a bounding box for the parent. Thanks for all the help. I guess the next steps should be either
- Add this in the documentation that parent objects should expect bounding box to be null 
- Log an error that Bounding box does not exist since this is a parent object

OR

- Create a bounding box based on children for an item that has children

Which branch should send the pull request on?

**Comment by jwoolston** [comment]
Do the third option - create the box based on children. Send the PR to `master`.

**Comment by jwoolston** [comment]
Closing as resolved via #1780
